codex/blockexchange/engine/engine.nim

```diff
@@ -514,14 +558,14 @@ proc taskHandler*(b: BlockExcEngine, task: BlockExcPeerCtx) {.gcsafe, async.} =
 
```

Comment:
Above where we compute `wantBlocks` it is done like this:

```nim
var wantsBlocks =
    task.peerWants.filterIt(it.wantType == WantType.WantBlock and not it.inFlight)
```

I think you have just made sure that `peerWants` only keeps `WantBlock` entries (you removed `peerCtx.peerWants.add(e)` from `WantHave`) in `wantListHandler` above, so, we should not need to filter it here anymore on `WantType.WantBlock` - this would be misleading even I guess.

Like this:

```nim
var wantsBlocks =
    task.peerWants.filterIt(not it.inFlight)
```